src/device-state/target-state.ts

```diff
@@ -85,33 +93,42 @@ export const update = async (
 		const endpoint = url.resolve(apiEndpoint, `/device/v2/${uuid}/state`);
 		const request = await getRequestInstance();
 
-		const params: Headers = {
-			json: true,
+		const params: any = {
+			headers: {
+				Authorization: `Bearer ${currentApiKey}`,
+				json: true,
+				['If-None-Match']:
+					typeof cache?.etag === 'string' ? cache.etag : undefined,
+			},
 		};
 
-		if (typeof cache?.etag === 'string') {
-			params.headers = {
-				'If-None-Match': cache.etag,
-			};
-		}
-
 		const [{ statusCode, headers }, body] = await request
-			.getAsync(endpoint, {
-				json: true,
-			})
+			.getAsync(endpoint, params)
 			.timeout(apiTimeout);
 
 		if (statusCode === 304) {
 			// There's no change so no need to update the cache or emit a change event
 			return;
 		}
 
+		if (statusCode < 200 || statusCode >= 300) {
+			log.error(`Error from the API: ${statusCode}`);
+			throw new ApiResponseError(`Error from the API: ${statusCode}`);
```

Comment:
Do we need to both log and throw here? Just wondering if it'll be caught and logged twice further up

Reply:
It isn't logged so it doesn't show up and the logs just look like nothing is wrong...

Reply:
Ok, just checking 👍